### PR TITLE
authWebview: UI buttons open auth webview

### PR DESCRIPTION
**Pull request by nkomonen-amazon**
## Problem
We want to direct users to the Add Connection webview

## Solution
Have the webview open in the places the mock shows in the image below:

## How to test
The new functionality is hidden behind a setting. If the setting is not set, the code
falls back to the original functionality, specifically `show.ts#executeFallbackLogic()`.
- set the setting `aws.dev.forceDevMode: true` in the vscode settings json
- Click around to try

Screen Shot 2023-06-14 at 10 45 51 AM

### No Connection Setup
Kapture 2023-06-14 at 14 35 52

### When existing connections
Kapture 2023-06-14 at 14 37 43

### After successful connection, button to go directly to relevant tree node
Kapture 2023-06-14 at 14 40 28

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by andrewyuq — APPROVED** [review]

File: src/auth/ui/vue/show.ts [hunk 1]
```diff
@@ -246,3 +301,24 @@ export async function showAuthWebview(ctx: vscode.ExtensionContext): Promise<voi
...
+ * logic if we are not in dev mode.
+ *
+ * TODO: Remove this dev mode check once our auth connections webview is fully implemented.
+ * We are currently doing this to fallback to the previous behaviour while still being
+ * able to pre-emptively update parts of the code to call the new functionality once
+ * things are finalized.
+ */
+function executeFallbackLogic(serviceToShow?: ServiceItemId) {
+    if (!DevSettings.instance.isDevMode()) {
+        if (serviceToShow === 'CODE_WHISPERER') {
```
[the remark this answers is not in the document]
Comment (justinmk3) [reply src/auth/ui/vue/show.ts:316]: Also don't think these ids need to be uppercase.

File: package.json [hunk 2]
```diff
@@ -1872,7 +1872,7 @@
             ],
             "aws.auth": [
                 {
-                    "command": "aws.auth.addConnection",
+                    "command": "aws.auth.showConnectionsPage",
```
Comment (justinmk3) [inline package.json:1875]: Similar to the "form" , no need to engrave the current delivery mechanism in the command name.

```suggestion
                    "command": "aws.auth.manageConnections",
```

File: src/auth/utils.ts [hunk 3]
```diff
@@ -43,14 +43,19 @@ import { Auth } from './auth'
 import { validateIsNewSsoUrl, validateSsoUrlFormat } from './sso/validation'
 import { openUrl } from '../shared/utilities/vsCodeUtils'
 
-export async function promptForConnection(auth: Auth, type?: 'iam' | 'sso') {
+// TODO: Look to do some refactoring to handle circular dependency later and move this to ./commands.ts
+export const showConnectionsPageCommand = 'aws.auth.showConnectionsPage'
```
Comment (justinmk3) [inline src/auth/utils.ts:47]: maybe `./src/shared/vscode/commands2.ts` is a place where these names could live

File: src/auth/ui/vue/types.ts [hunk 4]
```diff
@@ -4,3 +4,13 @@
  */
 
 export type ServiceItemId = 'DOCUMENT_TYPE_SUPPORT' | 'RESOURCE_EXPLORER' | 'CODE_WHISPERER' | 'CODE_CATALYST'
```
Comment (justinmk3) [inline src/auth/ui/vue/types.ts:6]: `ServiceItemId` as a name might have too much overlap with service ids found in AWS SDK (and the service models. 

This seems more like a "component id"?
```suggestion
export type toolkitComponent = 'doctypeTypes' | 'awsExplorer' | 'codewhisperer' | 'codecatalyst'
```

File: src/shared/treeview/resourceTreeDataProvider.ts [hunk 5]
```diff
@@ -107,6 +113,16 @@ export class ResourceTreeDataProvider implements vscode.TreeDataProvider<TreeNod
         return tracked
     }
 
+    public getParent(element: TreeNode<unknown>): vscode.ProviderResult<TreeNode> {
+        if (!element.getParent) {
+            getLogger().warn(
+                `Node '${element.id}' has not implemented getParent(). This can cause issues with vscode.TreeView.reveal().`
```
Comment (justinmk3) [inline src/shared/treeview/resourceTreeDataProvider.ts:119]: Maybe in devmode we should throw here

**Review by justinmk3 — APPROVED** [review]

File: src/shared/treeview/resourceTreeDataProvider.ts [hunk 6]
```diff
@@ -107,6 +113,15 @@ export class ResourceTreeDataProvider implements vscode.TreeDataProvider<TreeNod
         return tracked
     }
 
+    public getParent(element: TreeNode<unknown>): vscode.ProviderResult<TreeNode> {
+        if (!element.getParent) {
+            throw new ToolkitError(
```
Comment (justinmk3) [inline src/shared/treeview/resourceTreeDataProvider.ts:118]: Should this be done only for `aws.devMode` ?

**Pull request merged**